firestore-counter/clients/dart/distributed_counter.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:async/async.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
```

Comment:
```suggestion
import 'package:uuid/uuid.dart';
```
At the java version they used uuid as well. Unfortunately Dart doesn't have this, so I used a package. I am not sure if it is suitable for this package. So the way generating a random id might need to change.